core/arch/arm/dts/fsl-lx2160a.dtsi

```diff
@@ -525,6 +525,13 @@
 			little-endian;
 		};
 
+		sec_mon: sec-mon@1e90000 {
+			compatible = "fsl,lx2160a-sec-mon";
```

Comment:
Is this binding defined somewhere? I should really have asked when reviewing 3a34000547 but better late than never...

I don't think we discussed the location for the secure bindings in the past. In many cases, devices may be accessed either in secure or non-secure mode and therefore the bindings already exist in the Linux kernel tree (`Documentation/devicetree/bindings`). IMO, and ideally, even the secure-only drivers should have their bindings described there because it is the de-facto reference. See for example the comment in kernel patch that introduced  `secure-status` ([link](https://lkml.kernel.org/lkml/1448381201-10729-1-git-send-email-peter.maydell@linaro.org/T/)):
> This binding doesn't affect the kernel itself, but the kernel
> Documentation/ tree is the de-facto current place where all DT
> bindings are documented, so Grant suggested this was the right
> place to send a doc patch.

Here the `.dtsi` comes from the Linux kernel with minor changes, and thus I think this patch should also be sent to the LKML. Indeed we should try to minimize the changes with upstream or things could become difficult to maintain. On this topic, I can see that quite a few nodes are missing here. If, as I understand it, this DT is used by both OP-TEE and the kernel, then users of this file may be missing functionalities. It would probably a good idea to synchronize.

CC @sahilnxp @pangupta.

Reply:
Hey @jforissier 

Yes, I agree with you that we should have same DTS files being used both in kernel and OP-TEE, which will also reduce the maintenance.

@andrewmustea Since SecMon can be accessed by both Secure and Non Secure world, I think marking it to be used only from Secure World in `.dtsi` and upstreaming it will be more restrictive.
If your use case is like that you will access SecMon from Secure World only, I think you can make changes locally.

Reply:
> Is this binding defined somewhere? I should really have asked when reviewing [3a34000](https://github.com/OP-TEE/optee_os/commit/3a34000547dd3c274ad4f91592afb862c37ed4a8) but better late than never...
> 
> I don't think we discussed the location for the secure bindings in the past. In many cases, devices may be accessed either in secure or non-secure mode and therefore the bindings already exist in the Linux kernel tree (`Documentation/devicetree/bindings`). IMO, and ideally, even the secure-only drivers should have their bindings described there because it is the de-facto reference. See for example the comment in kernel patch that introduced `secure-status` ([link](https://lkml.kernel.org/lkml/1448381201-10729-1-git-send-email-peter.maydell@linaro.org/T/)):
> 
> > This binding doesn't affect the kernel itself, but the kernel
> > Documentation/ tree is the de-facto current place where all DT
> > bindings are documented, so Grant suggested this was the right
> > place to send a doc patch.
> 
> Here the `.dtsi` comes from the Linux kernel with minor changes, and thus I think this patch should also be sent to the LKML. Indeed we should try to minimize the changes with upstream or things could become difficult to maintain. On this topic, I can see that quite a few nodes are missing here. If, as I understand it, this DT is used by both OP-TEE and the kernel, then users of this file may be missing functionalities. It would probably a good idea to synchronize.
> 
> CC @sahilnxp @pangupta.

Hey @jforissier,

Should we wait to merge this PR until the upstream kernel is updated or should it be merged now?

Reply:
@andrewmustea well I suppose we can merge it.

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`